### ISSUE TITLE
[Android]: future proofing - HarmonyOS NEXT support

### ISSUE THREAD

**Issue by clang-clang-clang** [coronalabs/corona#649]
Solar2D may consider planning to support the "HarmonyOS NEXT" in case it cannot be built in the future.

> At the time, outlets such as [Huawei Central](https://www.huaweicentral.com/cutting-ties-with-android-harmonyos-next-is-ready-to-release-native-harmonyos-app-ecosystem/) said that “Next” is “a pure version of the HarmonyOS operating system without Android libraries” and that it would replace existing versions of the platform.

See: https://www.gsmarena.com/harmonyos_next_wont_have_android_app_support-news-60561.php and https://9to5google.com/2023/11/13/huawei-harmonyos-next-android/

**Comment by Shchvova** [comment]
Fortunately Solar2D is not dependent on Google’s libraries, only on open source android ones. I hope this will make it easier to port

**Comment by clang-clang-clang** [comment]
Updates.

> Note that HarmonyOS will [retain Android support internationally](https://www.gsmarena.com/huawei_will_keep_android_app_support_for_overseas_devices-news-60644.php), this mostly affects the home market.

See: [GSMARENA - HarmonyOS Next gets closer to prime time, video shows off the new UI design language](https://www.gsmarena.com/harmonyos_next_gets_closer_to_prime_time_video_shows_off_the_new_ui_design_language-news-61278.php)

**Comment by clang-clang-clang** [comment]
The beta version will be available in June 2024.

See: [Huawei to launch HarmonyOS NEXT beta at HDC 2024 event in June](https://www.huaweicentral.com/huawei-to-launch-harmonyos-next-beta-at-hdc-2024-event-in-june/)

**Comment by labolado** [comment]
There is a HarmonyOS port version: [https://gitee.com/fengozl/corona](https://gitee.com/fengozl/corona)

**Comment by sekodev** [comment]
Any chance this could be merged to the main repo?

**Comment by clang-clang-clang** [comment]
Not sure it will be useful to Solar2D users without additional information, maybe we have to wait until after the June HDC 2024 release?

**Comment by sekodev** [comment]
Of course. I was asking for future releases, after the end of beta maybe?

**Comment by clang-clang-clang** [comment]
Seems the HarmonyOS development kit is now available.
